### ISSUE TITLE
[$250] Screen Readers Sign in: The 'Language' dropdown has no role, name and state announced

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77556]
## Action Performed:
Prerequisites:
- the user is not signed in

1. Using iOS VoiceOver or Android TalkBack, open the Expensify app
2. Swipe to the 'Language' dropdown in the footer
3. Note how the control is announced.

## Expected Result:
The control should be announced with its role, name, value and state, e.g.:
- 'Select Language <option>, button, collapsed'

## Actual Result:
The control is announced without its name, role and state, nothing announces on focus.

## Area issue was found in:
Sign in

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Ensure the element is implemented as a button.
https://reactnative.dev/docs/accessibility#accessibilityrole

Use the accessibilityLabel to announce the name and selected option.
https://reactnative.dev/docs/accessibility#accessibilitylabel

Make sure the collapsed or expanded state is announced.
https://reactnative.dev/docs/accessibility#accessibilitystate

## Environment:
Apple iPhone 11 - iOS 17.5.1, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos

<img width="828" height="1792" alt="Image" src="https://github.com/user-attachments/assets/6b3a9124-5540-4837-8cd5-e4ecbd10d2a8" />

https://github.com/user-attachments/assets/33916202-3ddd-4347-a05d-1cd7d57a6ed3

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7005070
**Issue reported by:** NA
**Slack conversation:** NA
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~022013103989389854958</li>
        <li>Upwork Job ID: 2013103989389854958</li>
        <li>Last Price Increase: 2026-01-19</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @JS00001</details>

**Comment by rushatgabhane** [comment]
I couldn't fix this using AI.

Looking for a proposal : )

**Comment by Ollyws** [comment]
Although there is the `accessibilityState` state prop for this purpose it seems unreliable on iOS and I can't get it to work. Therefore I think @nkdengineer's [solution](https://github.com/Expensify/App/issues/77556#issuecomment-3766337448) of just creating our own accessability label with the state of the dropdown is the way to go here.

🎀👀🎀 C+ reviewed

**Comment by Uzaifm127** [comment]
@JS00001 That issue is unassigned but that issue is a part of this [parent issue](https://github.com/Expensify/App/issues/79866) and in that parent issue we have two more issues to solve.

So I thought like we should solve it in bulk rather than treating it as separate.

**Comment by Ollyws** [comment]
@nkdengineer You can test on the simulator using XCode accesibility inspector. XCode > Open Developer Tool > Accesibility inspector

**Comment by twisterdotcom** [comment]
@nkdengineer has submitted payment for this report on ND [here](https://staging.new.expensify.com/search?q=sortBy:date%20sortOrder:desc%20type:expense%20groupBy:reports%20reportID:R00DwCfeHu8R%20policyID:9132779747441AFB).

SO: https://stackoverflowteams.com/c/expensify/questions/7582
- [x] Vendor found in [VML](https://docs.google.com/spreadsheets/d/1Fs-umMmHhCCMzxiczyNydl9Rd6jdO0pQskYA3RvNWQc/edit?gid=1740618654#gid=1740618654): True. Status: Active
- [x] Amount requested: $250.00 (Matches the payment summary here: https://github.com/Expensify/App/issues/77556#issuecomment-4163458898.)
- [x] I, @twisterdotcom am not already the BZ assignee. I have approved this payment.

<details><summary>Zapier Logs</summary>
<b>Run ID:</b> <a href="https://zapier.com/editor/311565481/run/00040eee-fde9-a237-82f6-506bcbf99e04/run-details">00040eee-fde9-a237-82f6-506bcbf99e04</a>
</details>
